### PR TITLE
Web UI development workshop

### PR DESCRIPTION
**Pull request by Diksha-Verma-Dev**
Web UI development workshop: https://nuxeowiki.atlassian.net/wiki/spaces/NuxEng/pages/3244458028/Web+UI+development+workshop

### CODE REVIEW

**Review by Gabez0r — COMMENTED** [review]
Well done! 👏

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 1]
```diff
@@ -151,6 +165,10 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
           type: Object,
           observer: '_setBreadcrumbElements',
         },
+        icon: {
+          type: String,
+          value: 'nuxeo:delete',
+        },
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: There's no requirement to expose this as a public property. Every property we expose is public API then we will have to maintain in the future. We can perfectly inline the icon name in the element.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 2]
```diff
@@ -125,14 +149,28 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                      class="current breadcrumb-item breadcrumb-item-current"
+                      aria-current="page"
+                      title="[[_title(document)]]"
+                    >
+                      [[_title(document)]]
+                    </a>
+                  
+                        <span class="doc-uid">
+                        [[_documentUID(document)]]
+                      </span>
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: The indentation seems to be a bit off here.

**Review by Gabez0r — COMMENTED** [review]
Good job. Please check comments under https://github.com/nuxeo/nuxeo-web-ui/pull/1454#pullrequestreview-1015043759.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 3]
```diff
@@ -125,14 +149,29 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                      href$="[[urlFor(document)]]"
+                      class="current breadcrumb-item breadcrumb-item-current"
+                      aria-current="page"
+                      title="[[_title(document)]]"
+                    >
+                      [[_title(document)]]
+                    </a>
+                       <span class="doc-uid">
+                        [[_documentUID(document)]]
+                       </span>
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: Indentation seems to be a bit off here.

File: elements/nuxeo-browser/nuxeo-breadcrumb.js [hunk 4]
```diff
@@ -125,14 +149,29 @@ import { microTask } from '@polymer/polymer/lib/utils/async.js';
...
+                </div>
+                <template is="dom-if" if="[[_isTrashed(document)]]">
+                  <span class="trash-icon-parent">
+                   <paper-icon-button icon="nuxeo:delete"
```
Comment (Gabez0r) [inline elements/nuxeo-browser/nuxeo-breadcrumb.js]: Note that there's no good reason to user a button here if we are not expecting the user to click on it. We should use the [iron-icon](https://www.webcomponents.org/element/@polymer/iron-icon) instead, which already exposes [a couple of CSS variables to set width and height](https://github.com/PolymerElements/iron-icon/blob/master/iron-icon.js#L79-L80).

**Review by Gabez0r — COMMENTED** [review]
Good work. Please see comments for:
- development: https://github.com/nuxeo/nuxeo-web-ui/pull/1454#pullrequestreview-1024500921
- functional tests: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1024516164

File: elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-tooltip.js';
+import '../document/nuxeo-document-form-layout.js';
+import '@polymer/paper-button/paper-button.js';
+
+/**
+`nuxeo-document-parent-inspector-button`
+@group Nuxeo UI
+@element nuxeo-document-parent-inspector-button
+*/
+Polymer({
```
Comment (richardsd) [inline elements/nuxeo-document-actions/nuxeo-document-parent-inspector-button.js]: Please refer to https://github.com/nuxeo/nuxeo-web-ui/pull/1454#discussion_r910797179.

File: elements/nuxeo-web-ui-bundle.html [hunk 6]
```diff
@@ -231,6 +231,16 @@
 
 <!-- DOCUMENT ACTIONS -->
 
+<nuxeo-slot-content name="parentInspectorDocumentAction" slot="DOCUMENT_ACTIONS" order="10">
+  <template>
+    <nuxeo-filter user="[[user]]" expression="user.isAdministrator || this.isMember(user, 'powerusers')">
```
Comment (richardsd) [inline elements/nuxeo-web-ui-bundle.html]: As an alternative solution, please check https://github.com/nuxeo/nuxeo-elements/blob/maintenance-3.0.x/ui/nuxeo-filter.js#L167. nuxeo-filter has a group property that can be use to check user membership to groups.

File: packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  this.ui.browser.parentInspectorDataUID === 'UID:';
+  this.ui.browser.parentInspectorDataFacets === 'Facets:';
+  this.ui.browser.parentInspectorDataSchema === 'Schemas:';
+});
+
+Then('I can see dialog showing some information for a given document', function() {
+  this.ui.browser.parentInspectorDataTitle === 'Title:';
+  this.ui.browser.parentInspectorDataPath === 'Path:';
+  this.ui.browser.parentInspectorDataFacets === 'Facets:';
+  this.ui.browser.waitForNotVisible(this.ui.browser.parentInspectorDataUID);
```
Comment (richardsd) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js]: Similar to what you are doing here to check that some fields are not visible, you should check that other fields are displayed.

File: packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import pause from 'webdriverio/build/commands/browser/pause';
+
+Then('I can open current document information', function() {
+  this.ui.browser.parentInspectorButton.waitForVisible();
+  this.ui.browser.parentInspectorButton.click();
+  this.ui.browser.parentInspectorDialog.waitForVisible();
+});
+
+Then('I can see dialog showing the information for a given document', function() {
+  this.ui.browser.parentInspectorDataTitle === 'Title:';
```
Comment (richardsd) [inline packages/nuxeo-web-ui-ftest/features/step_definitions/parent_inspector.js]: Please see my comment bellow.

**Review by Gabez0r — COMMENTED** [review]
Please don't forget to update your element to class-based syntax. See more comments here: Good job. Please see comments here: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1027608150.

**Review by Gabez0r — COMMENTED** [review]
Well done! Please see comments here: https://github.com/nuxeo/nuxeo-web-ui/pull/1491#pullrequestreview-1028279090

**Pull request closed without merging**